docs/cli/ilab-rag-retrieval.md

```diff
@@ -0,0 +1,276 @@
+# Design Proposal - Embedding Ingestion Pipeline And RAG-Based Chat
+**TODOs**: 
+* Vector store authentication options.
+* Document versioning and data update policies.
+* Unify prompt management in InstructLab. See (`chat_template` [configuration][chat_template] and
+`augment_chat_template` [function][augment_chat_template])
+
+**Author**: Daniele Martinoli
+
+**Version**: 0.1
+
+## 1. Introduction
+This document proposes enhancements to the `ilab` CLI to support workflows utilizing Retrieval-Augmented Generation
+(RAG) artifacts within `InstructLab`. The proposed changes introduce new commands and options for the embedding ingestion
+and RAG-based chat pipelines:
+* A new `ilab data` sub-command to process customer documentation.
+* A new `ilab data` sub-command to generate and ingest embeddings from pre-processed documents into a configured vector store.
+* An option to enhance the chat pipeline by using the stored embeddings to augment the context of conversations, improving relevance and accuracy.
+
+## 2. Proposed Pipelines
+### 2.1 Working Assumption
+This proposal aims to serve as a reference design to develop a Proof of Concept for RAG workflows, while
+also laying the foundation for future implementations of state-of-the-art RAG artifacts tailored to specific use
+cases.
+
+#### Command Options
+To maintain compatibility and simplicity, no new configurations will be introduced for new commands. Instead,
+the settings will be defined using the following hierarchy (options higher in the list overriding those below):
+* CLI flags (e.g., `--FLAG`).
+* Environment variables following a consistent naming convention, such as `ILAB_<UPPERCASE_ARGUMENT_NAME>`.
+* Default values, for all the applicable use cases.
+
+For example, the `vectordb-uri` argument can be implemented using the `click` module like this:
+```py
+@click.option(
+    "--vectordb-uri",
+    default='rag-output.db',
+    envvar="ILAB_VECTORDB_URI",
+)
+```
+
+#### Local embedding models
+The embedding model used to generate text embeddings must be downloaded locally before executing the pipeline.
+
+For example, this command can be used to download the `sentence-transformers/all-minilm-l6-v2` model to the local models cache:
+```bash
+ilab model download -rp sentence-transformers/all-minilm-l6-v2
+```
+
+If the configured embedding model has not been cached, the command execution will terminate with an error. This requirement applies
+consistently to all new and updated commands.
+
+### 2.2 Document Processing Pipeline
+The proposal is to add a `process`  sub-command to the `data` command group:
+```
```

Comment:
Review the command to fetch, by default, knowledge definitions from the taxonomy repo.
`--input` and `--output` flags are optional arguments instead.